include/seastar/core/iostream-impl.hh

```diff
@@ -196,13 +201,8 @@ input_stream<CharType>::read_exactly(size_t n) noexcept {
             return this->read_exactly(n);
         });
     } else {
-      try {
         // buffer too small: start copy/read loop
-        tmp_buf b(n);
-        return read_exactly_part(n, std::move(b), 0);
-      } catch (...) {
```

Comment:
this exception-handling machinery can be removed, right? (If I understand correctly, it serves no purpose now that the allocation of the tmp_buf has been moved into `read_exactly_part`.

Reply:
Right, and since coroutines automatically wrap exceptions in futures.